polkadot/node/subsystem-util/src/inclusion_emulator/mod.rs

```diff
@@ -750,7 +764,7 @@ fn validate_against_constraints(
 		})
 	}
 
-	if commitments.upward_messages.len() > constraints.max_ump_num_per_candidate {
+	if upward_message_count > constraints.max_ump_num_per_candidate {
```

Comment:
we can access the `ump_messages_sent` from the `modifications` param and spare this extra parameter